src/coreclr/src/jit/hwintrinsic.h

```diff
@@ -658,14 +701,28 @@ struct HWIntrinsicInfo
         HWIntrinsicFlag flags = lookupFlags(id);
         return (flags & HW_Flag_SpecialImport) != 0;
     }
+
+#ifdef TARGET_ARM64
+    static bool SIMDScalar(NamedIntrinsic id)
+    {
+        const HWIntrinsicFlag flags = lookupFlags(id);
+        return (flags & HW_Flag_SIMDScalar) != 0;
```

Comment:
Why a flag, rather than a category?

Reply:
Because, many intrinsics can have features of SIMDScalar and some other category:
1) SIMD ShiftLeftByImmediate + SIMDScalar
2) SIMD ShiftRightByImmediate + SIMDScalar
3) SIMD ByIndexedElement + SIMDScalar

and I didn't want to have all possible combinations of those.

In addition, on Arm64 flag SIMDScalar is going to be used only in CodeGen to set INS_OPTS_NONE (i.e. switch between vector variant to scalar variant for a given instruction)
while a category value is used in multiple phases (e.g.  ByIndexedElement is used in LSRA, Lower and Importer) for making various decisions (what registers can be used for an indexed element, how to perform containment analysis etc.).

Reply:
Thanks! Makes sense to me.